### ISSUE TITLE
Detect @Inject constructors on framework components

### ISSUE THREAD

**Issue by jzbrooks** [google/dagger#3749]
Fragments, Activities, etc should use @AndroidEntryPoint to perform member injection. @Inject constructs won't work here since the framework instantiates the components and dagger has no opportunity to provide constructor parameters.

The main value of this check is an explanation to engineers new to dagger why this is wrong. A quick fix should be doable for this. The detector and fix should work for both Java and Kotlin source.

The hilt jar should provide this functionality if a hilt assumption is baked into the detector and fix.

**Comment by jzbrooks** [comment]
Similarly to #3729, we may build this in our lint library for now. I see a lot of value in upstreaming, however.

**Comment by JakeWharton** [comment]
Note that you can constructor-inject activities and fragments with an app component factory and no Hilt: https://developer.android.com/reference/android/app/AppComponentFactory

**Comment by jzbrooks** [comment]
Yep. That's why I'd rather the fix go into the hilt jar instead of a generic rule set like https://github.com/Faithlife/AndroidLint, that way hilt (love or hate it) use can be assumed.
